packages/core/src/odata/common/entity.ts

```diff
@@ -252,11 +257,26 @@ export abstract class EntityBase {
     return Object.keys(this)
       .filter(key => this.propertyIsEnumerable(key))
       .reduce(
-        (accumulatedMap, key) => ({ ...accumulatedMap, [key]: this[key] }),
+        (accumulatedMap, key) => ({
+          ...accumulatedMap,
+          [key]: this.getCurrentStateForKey(key)
+        }),
         this.getCustomFields()
       ) as this;
   }
 
+  protected getCurrentStateForKey(key: string) {
```

Comment:
what if I have a circular reference between two entites? Persons ahs Friends and each Friend has a reference back to the Person. Could this lead to an infinite loop?

Reply:
Also I am confused by the name `CurrentMapKeys`. As I see it build the full object following all navigation properties and doing this from the local state not the remote one right? How is this different to the entity in object notation leaving undfined values?

Reply:
I don't think so. This would be the case if we had this situation:
```ts
const parent = {};
const child = { parent };
parent.child = child;
```
Here the objects are set as references to one another. I think in our case we rather get something like this:
```ts
const apiParentData = {
  someSelectedParentProperty: 'test',
  child: {
    someSelectedChildProperty: 'test',
    // only here if expanded
    parent: {
       someSelectedParentProperty: 'test',
       // only here if expanded
       child: ...
    }
  }
}
```
I would argue that in this example the data is not referencing itself. If you want to achieve this endlessly you would have to endlessly request the data.

Not sure if you can achieve this through the api though.

Reply:
I made a test. Yeah. it's possible...